### ISSUE TITLE
feat: Add Option to Permanently Delete Email Conversations ChatWoot

### ISSUE THREAD

**Issue by nikul-tc** [chatwoot/chatwoot#10503]
### Is your feature or enhancement related to a problem? Please describe.

When deleting an email conversation in ChatWoot on a self-hosted instance (e.g., DigitalOcean 1-Click setup), the conversation is re-imported on the next polling cycle if it still exists in the IMAP server. This frustrates managing email conversations directly within ChatWoot, especially for spam or unwanted emails. Users are forced to delete the emails directly from their IMAP server (e.g., Gmail), which could be better for efficient workflow management.

### Describe the solution you'd like

Introduce an option to delete email conversations within ChatWoot permanently. When this option is selected:

The conversation is deleted from ChatWoot.
Optionally, the corresponding email is deleted or moved to a specified folder (e.g., Trash) in the connected IMAP server.
This feature would allow for better management of email conversations, enabling users to keep their ChatWoot environment clean without the need to access their IMAP server manually, thus enhancing their productivity.

### Describe alternatives you've considered

Moving the emails to a separate IMAP folder manually could be more convenient and time-consuming for managing multiple conversations.
Adjusting IMAP polling or configuration settings doesn’t address the core issue of re-importing deleted emails into ChatWoot.

### Additional context

Adding an option to flag emails as spam or unknown and preventing re-imports would be helpful.
This feature would improve usability for teams handling large volumes of email conversations.

**Comment by jsmizmaul** [comment]
We solved this issue using Microsoft Flow - https://flow.microsoft.com - Power Automation. 
You can add a "delay" rule that will wait like a minute or so before moving new emails to archive. Once that is moved, you are free to delete the email from chatwoot. Its just an work around, I would love to see a native option from Chatwoot.

**Comment by bkonia** [comment]
I agree, this needs to be addressed. In addition to the option to delete from the server when deleted from Chatwoot, there should be a few additional options:

1. **Delete from the server when initially retrieved**. Once an email has been successfully retrieved, it would be deleted from the server. This would save resources, since Chatwoot would not have to scan through thousands of emails each time, to determine which emails need to be retrieved.
2. **Move to archive when initially retrieved**. If this option is enabled, Chatwoot would move each email to a designated archive folder after it's successfully retrieved. If it's subsequently deleted in Chatwoot, it would also be deleted from the archive folder.
3. **Move to Spam folder** - We need a Mark as Spam option in Chatwoot. This would remove the conversation from Chatwoot and move the email to the designated Spam folder on the server.

**Comment by wiredhosting** [comment]
No progress on this yet?

Thanks

**Comment by sojan-official** [comment]
Thanks for the reports and the additional context here.

We are aware of this issue and are working on a fix for the IMAP re-import behavior. The direction we are looking at is to make Chatwoot remember emails that were already fetched or intentionally removed from Chatwoot, so deleting a conversation does not cause the same email to be imported again on the next sync.

We do not plan to delete emails from the source mailbox by default. Any mailbox-side action, such as moving or deleting emails on the mail server, would need to be an explicit setting.
